src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs

```diff
@@ -43,6 +45,8 @@ public void Load(IEnumerable<string> paths)
                     }
                 }
             }
+
+            _assemblyRelatedExtensions = new ConcurrentDictionary<string, string>();
```

Comment:
In current form your dictionary treat `lib/net472/A.B.C.dll` vs `LIB/net472/A.B.C.dll` differently. Does is matter in some cases?

Reply:
Thanks for reviewing!
The key is the assets path (and it only generated for assembly type)
The assets path is assigned at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItemCollection.cs#L32
And the ContentItemCollection is the value of dictionary [_contentItems](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilderCache.cs#L24), and the key is a specific package (id + version + sha512). So for the same specific package, when loading the paths, it should always get the constant value, so there won't be any chance to get inconsistent paths.